### ISSUE TITLE
[Bug] Password reset link is missing in an email sent using SendGrid on self-hosted version

### ISSUE THREAD

**Issue by ajinkyakulkarni** [appsmithorg/appsmith#5453]
## Description

A user on Discord [reported](https://discord.com/channels/725602949748752515/760761686549463060/859081527304519680) that the password link (href attrib on <a> tag) is missing from the email received.

![emailmissing](https://user-images.githubusercontent.com/800578/123657899-dc4cc700-d84e-11eb-8426-77d41a2d59a7.png)

### Important Details

- Version: Self-Hosted with SendGrid

**Comment by rubabuddin** [comment]
please let me know if there is a resolution - happy to hop on a zoom call to debug/screenshare

**Comment by sharat87** [comment]
@rubabuddin, hey, thanks for letting us know about this issue. Can you confirm if this was on https://app.appsmith.com or if this was on a self-hosted installation please? Also, did you originally signup using email/password form or using one of the OAuth buttons?

**Comment by rubabuddin** [comment]
@sharat87 originally signed up using email/password form on the self-hosted installation

**Comment by sharat87** [comment]
hey @rubabuddin, I've setup a new self-hosted instance on an EC2 and tried to replicate this, but do see the URL set for the "Reset password" button.

![Screenshot 2021-07-02 at 09 35 16](https://user-images.githubusercontent.com/120119/124218915-d68a0680-db18-11eb-8443-1e5d1dc43ffb.png)

I tried looking into how this was implemented in the code to see if I can get a clue there. It looks like the URL is built using the value of the `Origin` header sent from the browser. So, if there's a load balancer (or some other proxying layer) between the browser and the Appsmith server, it should be forwarding the `Origin` header ahead. If it is not doing so, Appsmith server won't be able to infer the base URL of the application and so this error might occur. Can you confirm if you are using any load balancer over your server and, if yes, if it is configured to forward the `Origin` header?

**Comment by rubabuddin** [comment]
@sharat87 hmm not seeing any load balancer set up - do I need to set the `APPSMITH_CLOUD_SERVICES_BASE_URL` within the docker.env file?

**Comment by mohanarpit** [comment]
@rubabuddin No, you don't need to configure that URL. 

Would you be willing to get on a quick call to debug this? If yes, I'd request you to please book a convenient slot via my [Calendly link](https://calendly.com/arpit-appsmith/debugging-session?back=1&month=2021-07)

**Comment by mohanarpit** [comment]
While debugging this issue with the user, we figured that Gmail doesn't allow link clicks for emails that are landing in Spam. Since the user's email provider is not yet verified, Gmail is treating all the mails originating from that address as spam. Once we marked the email as explicitly "Not Spam", the message moved to the Inbox and the links started to work. 

Closing this issue as it's resolved.
